homeassistant/components/lg_thinq/sensor.py

```diff
@@ -0,0 +1,590 @@
+    )
+}
+PREFERENCE_SENSOR_DESC: dict[ThinQProperty, SensorEntityDescription] = {
+    ThinQProperty.RINSE_LEVEL: SensorEntityDescription(
+        key=ThinQProperty.RINSE_LEVEL,
+        translation_key=ThinQProperty.RINSE_LEVEL,
+    ),
+    ThinQProperty.SOFTENING_LEVEL: SensorEntityDescription(
+        key=ThinQProperty.SOFTENING_LEVEL,
+        translation_key=ThinQProperty.SOFTENING_LEVEL,
+    ),
+    ThinQProperty.MACHINE_CLEAN_REMINDER: SensorEntityDescription(
+        key=ThinQProperty.MACHINE_CLEAN_REMINDER,
+        translation_key=ThinQProperty.MACHINE_CLEAN_REMINDER,
+    ),
+    ThinQProperty.SIGNAL_LEVEL: SensorEntityDescription(
+        key=ThinQProperty.SIGNAL_LEVEL,
+        translation_key=ThinQProperty.SIGNAL_LEVEL,
+    ),
+    ThinQProperty.CLEAN_LIGHT_REMINDER: SensorEntityDescription(
+        key=ThinQProperty.CLEAN_LIGHT_REMINDER,
+        translation_key=ThinQProperty.CLEAN_LIGHT_REMINDER,
+    ),
+}
+RECIPE_SENSOR_DESC: dict[ThinQProperty, SensorEntityDescription] = {
+    ThinQProperty.RECIPE_NAME: SensorEntityDescription(
+        key=ThinQProperty.RECIPE_NAME,
+        translation_key=ThinQProperty.RECIPE_NAME,
+    ),
+    ThinQProperty.WORT_INFO: SensorEntityDescription(
+        key=ThinQProperty.WORT_INFO,
+        translation_key=ThinQProperty.WORT_INFO,
+    ),
+    ThinQProperty.YEAST_INFO: SensorEntityDescription(
+        key=ThinQProperty.YEAST_INFO,
+        translation_key=ThinQProperty.YEAST_INFO,
+    ),
+    ThinQProperty.HOP_OIL_INFO: SensorEntityDescription(
+        key=ThinQProperty.HOP_OIL_INFO,
+        translation_key=ThinQProperty.HOP_OIL_INFO,
+    ),
+    ThinQProperty.FLAVOR_INFO: SensorEntityDescription(
+        key=ThinQProperty.FLAVOR_INFO,
+        translation_key=ThinQProperty.FLAVOR_INFO,
+    ),
+    ThinQProperty.BEER_REMAIN: SensorEntityDescription(
+        key=ThinQProperty.BEER_REMAIN,
+        native_unit_of_measurement=PERCENTAGE,
+        translation_key=ThinQProperty.BEER_REMAIN,
+    ),
+}
+REFRIGERATION_SENSOR_DESC: dict[ThinQProperty, SensorEntityDescription] = {
+    ThinQProperty.FRESH_AIR_FILTER: SensorEntityDescription(
+        key=ThinQProperty.FRESH_AIR_FILTER,
+        translation_key=ThinQProperty.FRESH_AIR_FILTER,
+    ),
+    ThinQProperty.ONE_TOUCH_FILTER: SensorEntityDescription(
+        key=ThinQProperty.ONE_TOUCH_FILTER,
+        translation_key=ThinQProperty.ONE_TOUCH_FILTER,
+    ),
+}
+RUN_STATE_SENSOR_DESC: dict[ThinQProperty, SensorEntityDescription] = {
+    ThinQProperty.CURRENT_STATE: SensorEntityDescription(
+        key=ThinQProperty.CURRENT_STATE,
+        translation_key=ThinQProperty.CURRENT_STATE,
+    ),
+    ThinQProperty.COCK_STATE: SensorEntityDescription(
+        key=ThinQProperty.COCK_STATE,
+        translation_key=ThinQProperty.COCK_STATE,
+    ),
+    ThinQProperty.STERILIZING_STATE: SensorEntityDescription(
+        key=ThinQProperty.STERILIZING_STATE,
+        translation_key=ThinQProperty.STERILIZING_STATE,
+    ),
```

Comment:
🤨 
What does this do?

Reply:
In Water purifier>

R | CLEANING | Sterilization mode
-- | -- | --
NORMAL | Normal mode

Reply:
And the cock state?

Reply:
It's been updated internally.
cock(body) : Cleaning / Normal
sterilization : On/Off/Cancel